File: Programacion Modulo Compras/SAE-COMPRAS/CapaNegocio/NReportes.vb
Imports System.Collections.Generic
Imports System.Data
Imports CapaDatos
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Public Class NReportes
	Public Shared Function MostrarTop10Proveedores(parFechaInicial As [String], parFechaFinal As [String]) As DataTable
		Return New DReportes().MostrarTop10Proveedores(parFechaInicial, parFechaFinal)
	End Function

	Public Shared Function MostrarTop10Empleados(parFechaInicial As [String], parFechaFinal As [String]) As DataTable
		Return New DReportes().MostrarTop10Empleados(parFechaInicial, parFechaFinal)
	End Function

	Public Shared Function MostrarTop10Productos(parFechaInicial As [String], parFechaFinal As [String]) As DataTable
		Return New DReportes().MostrarTop10Productos(parFechaInicial, parFechaFinal)
	End Function

	Public Shared Function MostrarHistorialProveedor(parFechaInicial As [String], parFechaFinal As [String], parProveedor As [String]) As DataTable
		Return New DReportes().MostrarHistorialProveedor(parFechaInicial, parFechaFinal, parProveedor)
	End Function

	Public Shared Function MostrarHistorialEmpleado(parFechaInicial As [String], parFechaFinal As [String], parEmpleado As [String]) As DataTable
		Return New DReportes().MostrarHistorialEmpleado(parFechaInicial, parFechaFinal, parEmpleado)
	End Function

	Public Shared Function MostrarHistorialProducto(parFechaInicial As [String], parFechaFinal As [String], parProducto As [String]) As DataTable
		Return New DReportes().MostrarHistorialProducto(parFechaInicial, parFechaFinal, parProducto)
	End Function
End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaNegocio/NReportes.cs
using System;
using System.Collections.Generic;
using System.Data;
using CapaDatos;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CapaNegocio
{
    public class NReportes
    {
        public static DataTable MostrarTop10Proveedores(String parFechaInicial, String parFechaFinal)
        {
            return new DReportes().MostrarTop10Proveedores(parFechaInicial, parFechaFinal);
        }

        public static DataTable MostrarTop10Empleados(String parFechaInicial, String parFechaFinal)
        {
            return new DReportes().MostrarTop10Empleados(parFechaInicial, parFechaFinal);
        }

        public static DataTable MostrarTop10Productos(String parFechaInicial, String parFechaFinal)
        {
            return new DReportes().MostrarTop10Productos(parFechaInicial, parFechaFinal);
        }

        public static DataTable MostrarHistorialProveedor(String parFechaInicial, String parFechaFinal,
            String parProveedor)
        {
            return new DReportes().MostrarHistorialProveedor(parFechaInicial, parFechaFinal, parProveedor);
        }

        public static DataTable MostrarHistorialEmpleado(String parFechaInicial, String parFechaFinal,
            String parEmpleado)
        {
            return new DReportes().MostrarHistorialEmpleado(parFechaInicial, parFechaFinal, parEmpleado);
        }

        public static DataTable MostrarHistorialProducto(String parFechaInicial, String parFechaFinal,
            String parProducto)
        {
            return new DReportes().MostrarHistorialProducto(parFechaInicial, parFechaFinal, parProducto);
        }
    }
}